solidity/contracts/BurntPixArchives.sol

```diff
@@ -134,23 +204,36 @@ contract BurntPixArchives is LSP8CappedSupply {
             }
             burntArchives[archiveId] = archive;
         }
+        // TODO: emit event so we can update UI
     }
 
     function mintArchive(bytes32 archiveId, address to) public {
         Archive memory archive = burntArchives[archiveId];
-        require(archive.creator == msg.sender, "BurntPixArchives: Only the archive creator can mint the archive");
+        // todo: do you need to require that the archive id exists?
+        require(
+            archive.creator == msg.sender,
+            "BurntPixArchives: Only the archive creator can mint the archive"
+        );
         _mint(to, archiveId, true, "");
+        // todo: should we emit an event here too?
```

Comment:
I think _mint already does that @Ledespana. On chain it appears as a "transfer" from the zero address to the minter
<img width="1141" alt="Screenshot 2024-04-06 at 8 38 54 PM" src="https://github.com/yearone-io/burntpix-archives/assets/128236767/b04228e3-74df-4964-ab55-16f7c9528a0f">